common/request.go

```diff
@@ -61,19 +61,28 @@ type Request struct {
 }
 
 // NewRequest creates a new HTTP request
-func NewRequest(ctx context.Context, event *network.EventRequestWillBeSent, f *Frame, redirectChain []*Request, interceptionID string, allowInterception bool) *Request {
+func NewRequest(
+	ctx context.Context, event *network.EventRequestWillBeSent, f *Frame,
+	redirectChain []*Request, interceptionID string, allowInterception bool,
+) (*Request, error) {
 	documentID := cdp.LoaderID("")
 	if event.RequestID == network.RequestID(event.LoaderID) && event.Type == "Document" {
 		documentID = event.LoaderID
 	}
+
+	u, err := url.Parse(event.Request.URL)
+	if err != nil {
+		return nil, err
```

Comment:
Let's return a richer error message here, like:

```golang
return nil, fmt.Errorf("cannot parse URL: %w", err)
```

It will result in:
`cannot parse URL: parse ":": missing protocol scheme`

Reply:
Done in 7dc6ef7.